sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobInputStream.java

```diff
@@ -97,4 +99,14 @@ public BlobProperties getProperties() {
         return this.properties;
     }
 
+    /**
+     * Allows for encrypted blobs to use BlobInputStream correctly by using the non-encrypted blob length
+     */
+    private static Long adjustBlobLength(Long initialLength, Context context) {
+        if (context != null && context.getData(Constants.ADJUSTED_BLOB_LENGTH_KEY).isPresent()) {
+            return (Long) context.getData(Constants.ADJUSTED_BLOB_LENGTH_KEY).get();
```

Comment:
Should we be using `long` instead of `Long` here? Looking over the `StorageInputStream` constructor and `getBlobSize()` method, they both use/return `long`. So it probably makes sense to make it consistent.